aten/src/ATen/TensorUtils.h

```diff
@@ -78,4 +79,31 @@ void checkBackend(CheckedFrom c, at::ArrayRef<Tensor> t, at::Backend backend);
 void * maybe_data_ptr(const Tensor& tensor);
 void * maybe_data_ptr(const TensorArg& tensor);
 
+
+inline at::optional<at::Type*> unifyTypes(const Tensor& a, const Tensor& b) {
```

Comment:
nit: this should probably be a name like "resultType"; this function doesn't actually unify the types, that's the job of the dispatcher.